### PR TITLE
Use pycountry to retrieve language names translations

### PR DESCRIPTION
**Pull request by unho**
Use pycountry to retrieve language names translations

Partial fix for https://github.com/translate/pootle/issues/6218

### CODE REVIEW

File: translate/lang/data.py [hunk 1]
```diff
@@ -24,6 +24,7 @@
 import gettext
 import locale
 import os
+import pycountry
```
Comment (dwaynebailey) [inline translate/lang/data.py]: These and associated uses need to operate conditionally and just pass through if pycountry isn't available.

File: translate/lang/data.py [hunk 2]
```diff
@@ -425,18 +430,21 @@ def gettext_lang(langcode=None):
     language, or the system language if no language is specified.
     """
     if langcode not in iso639:
-        if not langcode:
+        kwargs = {
+            'domain': 'iso_639',
```
Comment (dwaynebailey) [inline translate/lang/data.py]: Not correct , the domains upstream are the same as pycountry

File: translate/lang/data.py [hunk 3]
```diff
@@ -446,18 +454,21 @@ def gettext_country(langcode=None):
     language, or the system language if no language is specified.
     """
     if langcode not in iso3166:
-        if not langcode:
+        kwargs = {
+            'domain': 'iso_3166',
```
Comment (dwaynebailey) [inline translate/lang/data.py]: Ditto domain, they should be exactly the same

**Comment by dwaynebailey** [comment]
lgtm with comments, not sure how to test

File: translate/lang/data.py (hunk 2 above)
Comment (unho) [inline translate/lang/data.py]: From `pycountry` docs:

> Locales are available in the pycountry.LOCALES_DIR subdirectory of this package. The translation domains are called isoXXX according to the standard they provide translations for.

In fact according to my testing this is not even true:

```pytb
In [55]: t=gettext.translation('iso639', localedir=pycountry.LOCALES_DIR, languages=['af'])
---------------------------------------------------------------------------
IOError                                   Traceback (most recent call last)
<ipython-input-55-df36c6546d2d> in <module>()
----> 1 t=gettext.translation('iso639', localedir=pycountry.LOCALES_DIR, languages=['af'])

/usr/lib/python2.7/gettext.pyc in translation(domain, localedir, languages, class_, fallback, codeset)
    482         if fallback:
    483             return NullTranslations()
--> 484         raise IOError(ENOENT, 'No translation file found for domain', domain)
    485     # Avoid opening, reading, and parsing the .mo file after it's been done
    486     # once.

IOError: [Errno 2] No translation file found for domain: 'iso639'

In [56]: t=gettext.translation('iso_639', localedir=pycountry.LOCALES_DIR, languages=['af'])
---------------------------------------------------------------------------
IOError                                   Traceback (most recent call last)
<ipython-input-56-30a7c1d77089> in <module>()
----> 1 t=gettext.translation('iso_639', localedir=pycountry.LOCALES_DIR, languages=['af'])

/usr/lib/python2.7/gettext.pyc in translation(domain, localedir, languages, class_, fallback, codeset)
    482         if fallback:
    483             return NullTranslations()
--> 484         raise IOError(ENOENT, 'No translation file found for domain', domain)
    485     # Avoid opening, reading, and parsing the .mo file after it's been done
    486     # once.

IOError: [Errno 2] No translation file found for domain: 'iso_639'

In [57]: t=gettext.translation('iso639-3', localedir=pycountry.LOCALES_DIR, languages=['af'])

In [58]:
```

Comment (unho) [inline translate/lang/data.py]: Without a virtualenv and with no `pycountry`, the domain is `iso_639`:

```pytb
In [16]: t=gettext.translation(domain='iso_639', languages=['af'])

In [17]: t=gettext.translation(domain='iso639', languages=['af'])
---------------------------------------------------------------------------
IOError                                   Traceback (most recent call last)
<ipython-input-17-29c7f6751c3a> in <module>()
----> 1 t=gettext.translation(domain='iso639', languages=['af'])

/usr/lib/python2.7/gettext.pyc in translation(domain, localedir, languages, class_, fallback, codeset)
    482         if fallback:
    483             return NullTranslations()
--> 484         raise IOError(ENOENT, 'No translation file found for domain', domain)
    485     # Avoid opening, reading, and parsing the .mo file after it's been done
    486     # once.

IOError: [Errno 2] No translation file found for domain: 'iso639'

In [18]: t=gettext.translation(domain='iso639-3', languages=['af'])
---------------------------------------------------------------------------
IOError                                   Traceback (most recent call last)
<ipython-input-18-314a88252b3d> in <module>()
----> 1 t=gettext.translation(domain='iso639-3', languages=['af'])

/usr/lib/python2.7/gettext.pyc in translation(domain, localedir, languages, class_, fallback, codeset)
    482         if fallback:
    483             return NullTranslations()
--> 484         raise IOError(ENOENT, 'No translation file found for domain', domain)
    485     # Avoid opening, reading, and parsing the .mo file after it's been done
    486     # once.

IOError: [Errno 2] No translation file found for domain: 'iso639-3'

In [19]:
```

**Comment by dwaynebailey** [comment]
lgtm, would prefer to see this live on a staging server somewhere, but good enough for now

**Pull request merged**